configure.ac

```
@@ -6340,6 +6340,21 @@ AS_VAR_IF([with_readline], [no], [
     # in readline as well as newer editline (April 2023)
     AC_CHECK_TYPES([rl_compdisp_func_t], [], [], [readline_includes])
 
+    # Some editline versions declare rl_startup_hook as taking no args, others
+    # declare it as taking 2.
+    AC_CACHE_CHECK([if rl_startup_hook takes arguments], [ac_cv_readline_rl_startup_hook_takes_args], [
+        AC_COMPILE_IFELSE(
+            [AC_LANG_PROGRAM([readline_includes]
+                [[extern int test_hook_func(const char *text, int state);]],
+                [[rl_startup_hook=test_hook_func;]])],
+            [ac_cv_readline_rl_startup_hook_takes_args=yes],
+            [ac_cv_readline_rl_startup_hook_takes_args=no]
+        )
+    ])
+    AS_VAR_IF([ac_cv_readline_rl_startup_hook_takes_args], [yes], [
+      AC_DEFINE([RL_STARTUP_HOOK_TAKES_ARGS], [1], [Define if rl_startup_hook takes arguments])
```

This define should have a `Py_` prefix, not a `RL_` prefix, as per the PEP-7 guidelines.

```suggestion
      AC_DEFINE([Py_STARTUP_HOOK_TAKES_ARGS], [1], [Define if rl_startup_hook takes arguments])
```

---

Changed, though I notice the other macros set in this section of configure.ac don't use a Py_ prefix.

---

Yes, this is a known issue, though it is recorded in the C API workgroup repo:

- https://github.com/capi-workgroup/problems/issues/46

---

Still, any new identifier or macro should follow the naming guideline.